daemon/gluster-blockd.c

```diff
@@ -363,22 +453,114 @@ gbMinKernelVersionCheck(void)
       distro, tptr, verStr.release);
 
  fail:
-  pclose(fp);
+  GB_FREE(distro);
 
   exit(EXIT_FAILURE);
 }
 
 
+static bool
+gbDependencyVersionCompare(int dependencyName, char *version)
+{
+  size_t vNum[VERNUM_BUFLEN] = {0, };
+  char *verStr;
+  int i = 0, j;
+  char *token, *tmp;
+  bool done = false;
+  bool ret = false;
+
+
+  if (GB_STRDUP(verStr, version) < 0) {
+    LOG("mgmt", GB_LOG_ERROR,
+        "gbDependencyVersionCompare: failed to strdup (%s)", strerror(errno));
+    return ret;
+  }
+
+  token = strtok(verStr, ".-");
+  while( token != NULL ) {
+    done = false;
+    for (j = 0; j < strlen(token); j++) { /* say if version is 2.1.fb49, parse fb49 */
+      if (isdigit(token[j])) {
+        vNum[i] = atoi(token);
+      } else {
+        tmp = token;
+        for(; *tmp; ++tmp) {
+          if (isdigit(*tmp)) {
+            vNum[i] = atoi(tmp);  /* feed 49 from fb49, so done = true */
+            done = true;
+            break;
+          }
+        }
+      }
+      if (done) {
+        break;
+      }
+    }
+    token = strtok(NULL, ".-");
+    i++;
+  }
+
+  switch (dependencyName) {
+  case TCMURUNNER:
+    if (DEPENDENCY_VERSION(vNum[0], vNum[1], vNum[2]) >= DEPENDENCY_VERSION(1, 1, 3)) {
```

Comment:
versions for components should probably be placed in a header file so that they can be found at a common place

Reply:
OK. Will push this to version.h